File: Swift/insertionsort/insertionsort_250000.swift
func insertionSort<T: Comparable>(_ array: inout [T]) {
    for i in 1..<array.count {
        var j = i
        while j > 0 && array[j] < array[j-1] {
            array.swapAt(j, j-1)
            j -= 1
        }
    }
}


var debug = false

insertionSort(&arr)
if (debug) {
    print(arr)
}

File: CSHARP/insertionsort/insertionsort_250000.cs
using System;

class InsertionSort{   

    public static bool debug = false;

    public static void insertionSort(int[] array)
    {
        int n = array.Length;

        for (int i = 1; i < n; i++)
        {
            int key = array[i];
            int j = i - 1;

            while (j >= 0 && array[j] > key)
            {
                array[j + 1] = array[j];
                j--;
            }

            array[j + 1] = key;
        }
    }

    public static void Main()
    {
        insertionSort(ARRAY);
        if(debug){
            for(int i = 0; i < ARRAY.Length; i++){
                Console.Write(ARRAY[i] + " ");
            }
        }
    }
}